### PR TITLE
[Refactor] replace `ZMGenericMessage` in `ZMAssetClientMessage`

### PR DESCRIPTION
**Pull request by David-Henner**
## What's new in this PR?

- Replace `ZMGenericMessage` by `GenericMessage` in `ZMAssetClientMessage`
- Support needed helper methods of `GenericMessage`
- Update tests

### CODE REVIEW

File: Source/Utilis/Protos/GenericMessage+Assets.swift [hunk 1]
```diff
@@ -151,3 +184,41 @@ extension WireProtos.Asset.RemoteData {
         }
     }
 }
+
+extension GenericMessage {
```
Comment (typfel) [inline Source/Utilis/Protos/GenericMessage+Assets.swift:188]: Doesn't it make more sense for these methods to be `mutating` in Swift protobuf?

File: Source/Utilis/Protos/GenericMessage+Helper.swift [hunk 2]
```diff
@@ -154,6 +154,35 @@ extension GenericMessage {
     }
 }
 
+
+extension GenericMessage {
+    var v3_isImage: Bool {
+        return assetData?.original.hasRasterImage ?? false
```
Comment (KaterinaWire) [inline Source/Utilis/Protos/GenericMessage+Helper.swift:160]: Did you remove the same implementation for `ZMGenericMessage`?

File: Source/Utilis/Protos/GenericMessage+Assets.swift [hunk 3]
```diff
@@ -151,3 +184,41 @@ extension WireProtos.Asset.RemoteData {
...
+        let asset = WireProtos.Asset(original: nil, preview: preview)
+        return updatedAsset(withAsset: asset)
+    }
+    
+    func updatedAsset(withUploadedOTRKey otrKey: Data, sha256: Data) -> GenericMessage? {
+        let asset = WireProtos.Asset(withUploadedOTRKey: otrKey, sha256: sha256)
+        return updatedAsset(withAsset: asset)
+    }
+    
+    func updatedAsset(withAsset asset: WireProtos.Asset) -> GenericMessage? {
```
Comment (KaterinaWire) [inline Source/Utilis/Protos/GenericMessage+Assets.swift:215]: We have such method:
`public mutating func update(asset: WireProtos.Asset) {...`
Maybe the rest we could do as mutated as well.

File: Source/Utilis/Protos/GenericMessage+Helper.swift (hunk 2 above)
Comment (David-Henner) [reply Source/Utilis/Protos/GenericMessage+Helper.swift:172]: done here 1668281

File: Source/Utilis/Protos/GenericMessage+Assets.swift (hunk 3 above)
Comment (David-Henner) [reply Source/Utilis/Protos/GenericMessage+Assets.swift]: done here 11f2694

File: Source/Utilis/Protos/GenericMessage+Assets.swift (hunk 1 above)
Comment (David-Henner) [reply Source/Utilis/Protos/GenericMessage+Assets.swift:190]: done 11f2694

**Pull request merged**